scipy/linalg/src/id_dist/src/idzp_asvd.f

```diff
@@ -159,9 +159,9 @@ subroutine idzp_asvd0(m,n,a,krank,list,proj,u,v,s,ier,
 c       for further documentation).
 c
         implicit none
-        integer m,n,krank,list(n),ier
+        integer m,n,krank,ier
         real*8 s(krank)
-        complex*16 a(m,n),u(m,krank),v(n,krank),
+        complex*16 a(m,n),u(m,krank),v(n,krank),list(n),
```

Comment:
Fixed:
```
../scipy/linalg/src/id_dist/src/idzp_asvd.f:122:72:

  122 |      1                    w(iui),w(ivi),w(isi),ier,w(icol),w(iwork))
      |                                                                        1
Warning: Type mismatch in argument 'list' at (1); passed COMPLEX(8) to INTEGER(4)
```

Reply:
This one is not correct it looks like, the warning just changes to `Warning: Type mismatch in argument 's' at (1); passed COMPLEX(8) to REAL(8)`.

Reply:
That is a different warning talking about line 163:

https://github.com/scipy/scipy/blob/6432ef2d9cc1f628ef64111bb63008f1cd794c9f/scipy/linalg/src/id_dist/src/idzp_asvd.f#L161-L165

What I fixed is the warning of `'list'` which was `COMPLEX` to `INTEGER`.

Reply:
I looked at it again, this is definitely not correct. The first usage of `list` in this routine is `call idz_copycols(m,n,a,krank,list,col)`; `list` in that routine is also integer. 

The problem with this code is deeper, `w(*)` is a work array declared as `complex*16` but used for storing data of different types. Example on line 83 of this file:
```
 call idzp_aid(eps,m,n,a,winit,krank,w(ilist),w(iproj))
```
The last two parameters are both `w`; the first one should be integer and the second one complex.

In general this come is extremely poorly written. I think I'll revert this change, merge the rest of the fixes, and then we can silence the remaining warnings.